### ISSUE TITLE
A bug in the AaronTools.geometry.Geometry.map_ligand() function

### ISSUE THREAD

**Issue by Senppoa** [QChASM/AaronTools.py#18]
Hello, dear developers of AaronTools,

I am using the python version of AaronTools to map new ligands on the TS template, but I encountered a bug of the map_ligand() function here.

The substrates of the reaction will rotate after the ligand mapped (changes in position), especially for cases where two substrates are separated. This error does not occur when the reaction substrate is a whole. Below are pictures of the bug I encountered.

1. Here is the situation where substrates (reactants) are a whole, no bug：
tsa1:
![tsa1](https://github.com/QChASM/AaronTools.py/assets/44549341/eb00823c-7da0-4a09-8872-6b356da25a9c)
tsa1 after mapping:
![tsa1_map](https://github.com/QChASM/AaronTools.py/assets/44549341/b1c8a2e1-777a-4481-93de-27f316cba003)
tsa1 and tsa1_mapped:
![tsa1-tsa1_amp](https://github.com/QChASM/AaronTools.py/assets/44549341/50d77319-176b-43a1-9e0f-f83a3571f5cd)

2. Here is the situation where reactants are separated, a substrates movement bug will occur：
ts1:
![ts1](https://github.com/QChASM/AaronTools.py/assets/44549341/078a7726-53cc-4ad5-aed7-af6463bfeece)
ts1 after mapping:
![ts1_map](https://github.com/QChASM/AaronTools.py/assets/44549341/5adc44f9-07a4-4eba-9452-fc6e0320a975)
ts1 and ts1_mapped:
![ts1-ts1_map](https://github.com/QChASM/AaronTools.py/assets/44549341/70aa9043-cbdf-426f-abc0-21ebe2946bf2)
bug here:
![rotation_bug](https://github.com/QChASM/AaronTools.py/assets/44549341/b8a26999-1a15-47af-93ed-31f31029da56)

That's all, the bug of this problem may be in geometry.py, but I can't solve it without your help. In addition, I also found that the "minimize" parameter in the map_ligand() function seems to be invalid, and when I set it to "True" and "False", there is no difference in the mapping results.

Looking forward to your reply, thank you!

**Comment by ajs99778** [comment]
Thanks for reporting this. There was an extra bit of code in `map_ligand` that helps remove steric clashing for monodentate ligands (AaronTools sees your substrates in the TSS are monodentate ligands). However, this code was also outside a `if minimize:` block. That code should now only run if `minimize=True`. This should fix the issue.

**Comment by Senppoa** [comment]
> Thanks for reporting this. There was an extra bit of code in `map_ligand` that helps remove steric clashing for monodentate ligands (AaronTools sees your substrates in the TSS are monodentate ligands). However, this code was also outside a `if minimize:` block. That code should now only run if `minimize=True`. This should fix the issue.

Thanks for your help, I have fixed this issue by making the `if minimize:` judgment before the `# rotate monodentate to relieve clashing`.

```
# rotate monodentate to relieve clashing
if minimize:
    for ligand in self.components:
        if len(ligand.key_atoms) == 1:
            targets = ligand.atoms
            key = ligand.key_atoms[0]
            if self.center:
                start = self.COM(self.center)
                end = key.coords
            else:
                start = key.coords
                end = self.COM(key.connected)
            axis = end - start
            self.minimize_torsion(targets, axis, center=key, increment=10)

    self.remove_clash()
    
    self.minimize()
```

Furthermore, I wonder whether this rotational function should be performed with the two substrates considered to be monodentate ligands kept in constant relative positions? So the structure of the substrates in the TS will not be changed.

**Comment by ajs99778** [comment]
I agree that it doesn't make sense to rotate anything other than the new ligand(s). After thinking about it, I I'm guessing the original intend was to rotate *new* monodentate ligands. For example, if you replace bipyridine with two pyridines, we want to rotate each of the pyridines so they aren't bumping into each other. I'll make the changes for this soon.
